lib/autocomplete-manager.coffee

```diff
@@ -188,66 +116,46 @@ class AutocompleteManager
 
     # No suggestions? Cancel autocompletion.
     return unless suggestions.length
+    @suggestionList?.changeItems(suggestions)
+    @showSuggestionList()
+
+  providersForScopes: (scopes) =>
+    return [] unless scopes?
+    return [] unless @scopes
+    providers = []
+    for scope in scopes
+      if @scopes[scope]?
+        providers = _.union(providers, @scopes[scope])
+    providers.push(@fuzzyProvider) unless _.size(providers) > 0
+    providers
 
-    unless @overlayDecoration?
-      marker = @editor.getLastCursor()?.getMarker()
-      @overlayDecoration = @editor?.decorateMarker(marker, { type: 'overlay', item: this })
-
-    # Now we're ready - display the suggestions
-    @changeItems suggestions
-
-    @setActive()
-
-  changeItems: (items) ->
-    @items = items
-    @emitter.emit 'did-change-items', items
-
-  onDidChangeItems: (cb) ->
-    @emitter.on 'did-change-items', cb
-
-  # Private: Focuses the hidden input, starts listening to keyboard events
-  setActive: ->
-    @addKeyboardInteraction()
-    @active = true
+  # Private: Gets called when the user successfully confirms a suggestion
+  #
+  # match - An {Object} representing the confirmed suggestion
+  confirm: (match) =>
+    return unless @editorHasFocus()
+    return unless match?.provider?
+    return unless @editor?
 
-  # Private: Gets called when the content has been modified
-  contentsModified: =>
-    delay = parseInt(atom.config.get "autocomplete-plus.autoActivationDelay")
-    if @delayTimeout
-      clearTimeout @delayTimeout
+    replace = match.provider.confirm(match)
+    @editor.getSelections()?.forEach (selection) -> selection?.clear()
 
-    @delayTimeout = setTimeout @runAutocompletion, delay
+    @hideSuggestionList()
 
-  # Private: Gets called when the cursor has moved. Cancels the autocompletion if
-  # the text has not been changed and the autocompletion is
-  #
-  # data - An {Object} containing information on why the cursor has been moved
-  cursorMoved: (data) =>
-    @cancel() unless data.textChanged
+    return unless replace
+    @replaceTextWithMatch(match)
+    @editor.getCursors()?.forEach (cursor) ->
```

Comment:
Yeah in that case it makes more sense because the column is actually modified. In this case it's the exact same values so it seems like a no-op.